File: java/rest-simple-exercise/src/main/java/com/example/restsimple/converter/LocalDateTimeAttributeConverter.java
package com.example.restsimple.converter;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

@Converter(autoApply = true)
public class LocalDateTimeAttributeConverter implements AttributeConverter<LocalDateTime, Long> {

    @Override
    public Long convertToDatabaseColumn(LocalDateTime locDateTime) {
        return locDateTime == null ? null : locDateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
    }

    @Override
    public LocalDateTime convertToEntityAttribute(Long sqlTimestamp) {
        return sqlTimestamp == null ? null : Instant.ofEpochMilli(sqlTimestamp).atZone(ZoneId.systemDefault()).toLocalDateTime();
    }
}

File: kotlin/rest-simple-exercise/src/main/kotlin/com/example/restsimple/converter/LocalDateTimeAttributeConverter.kt
package com.example.restsimple.converter

import java.time.Instant
import java.time.LocalDateTime
import java.time.ZoneId
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class LocalDateTimeAttributeConverter : AttributeConverter<LocalDateTime?, Long?> {
    override fun convertToDatabaseColumn(locDateTime: LocalDateTime?): Long? {
        return locDateTime?.atZone(ZoneId.systemDefault())?.toInstant()?.toEpochMilli()
    }

    override fun convertToEntityAttribute(sqlTimestamp: Long?): LocalDateTime? {
        return if (sqlTimestamp == null) null else Instant.ofEpochMilli(sqlTimestamp).atZone(ZoneId.systemDefault())
            .toLocalDateTime()
    }
}
